### DISCUSSION TITLE
大佬们，群晖安装的套件，docker功能报错。这个怎么解决啊。

### DISCUSSION THREAD

**Discussion by WJMA-GIT** [hslr-s/sun-panel#215] [General]
Docker 出错了: Docker ERROR- [permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/json?all=1&size=1": dial unix /var/run/docker.sock: connect: permission denied]

用这个解决过，但是重启就没了。有没有啥法子直接给sunpan权限的命令
```
chmod 666 /var/run/docker.sock
```

**Comment by caijaj2041** [comment]
> Docker 出错了： Docker ERROR- [permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock： Get “http：//%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/json？all=1&size=1”： dial unix /var/run/docker.sock： connect： permission denied]
> 
> 用这个解决过，但是重启就没了。有没有啥法子直接给sunpan权限的命令
> 
> ```
> chmod 666 /var/run/docker.sock
> ```

跟你一样，请问您解决了吗

**Reply by WJMA-GIT** [reply]
使用docker版本安装了。但是我有一个思路，你可以去试试，可以的话记得回来说一下，感谢
去套件中心下载权限管理器，激活后在套件中勾选sun-panel，然后重启套件试试

**Reply by caijaj2041** [reply]
> 使用docker版本安装了。但是我有一个思路，你可以去试试，可以的话记得回来说一下，感谢 去套件中心下载权限管理器，激活后在套件中勾选sun-panel，然后重启套件试试

我也已经解决了，这个方法是作者提供的，可以解决
![c21aaf42915c91e8a2514bc358fb0370](https://github.com/user-attachments/assets/a6423805-af57-48ca-a396-ac88bef2027e)

**Reply by 577fkj** [reply]
> 使用docker版本安装了。但是我有一个思路，你可以去试试，可以的话记得回来说一下，感谢 去套件中心下载权限管理器，激活后在套件中勾选sun-panel，然后重启套件试试

![image](https://github.com/user-attachments/assets/9be85927-ee2b-4e2b-a77a-907370ae67ec)
![image](https://github.com/user-attachments/assets/1dd28d9c-b86f-493c-be3e-5d3431488adf)
![image](https://github.com/user-attachments/assets/37a7b513-d4a6-4e16-ba75-d415056ef5b7)
![image](https://github.com/user-attachments/assets/aa8c2820-d10b-4ceb-a24d-069d787fe15e)
用权限管理器并不行

**Comment by zealiao** [comment]
极空间 nas 也是出现这样的报错，看了教程的，希望能有解决方案。

**Comment by xjming1314** [comment]
我来当一个终结者：出现这个问题一般是两种情况，第一个没有配置路径映射。即没有按照作者说明，把/var/run/docker.sock映射到/var/run/docker.sock。  第二个是配置了映射，但是没权限。

1）路径映射有3种办法。 一个是通过群晖套件的社区版资源下载，这里是共享好了。 第二个是通过群晖的docker安装，但是ssh到后台，安装的时候自定义好映射路径。 第三个就是在群晖的docker已经配置好了，不想删除重建，然而可视化更改路径映射不能选择/var/run/docker.sock。

我是第三种。这里就要有点骚操作了。  第一步 ssh后，用这个命令，给docker.sock创建一个符号连接，类似于快捷方式的玩意。 ln -s /var/run/docker.sock /volume2/docker/homepage/。   此时去docker/homepage的后台就能看到docker.sock文件了。 但是群晖的可视化还是无法查看此文件，更不能选中。  

骚操作来了，先在homepag随便新建一个文件叫docker.sock（先把刚刚的符号连接重命名一下，命名为1），然后停止原先的容器，可视化选择文件映射，文件就用刚刚新建的这个docker.sock。  然后保存，不要运行容器。 保存完毕后，删除这个docker.sock，再把原始的docker.sock（被重命名为1的文件）再名字改回来。  改完启动容器即可。

2）配置完映射，还是报错，就是没权限。  那就sudo chmod 666 /var/run/docker.sock  给他个权限。  解决重启失效，就在群晖创建个触发任务，每次开机自动运行这个命令即可

**Reply by OnismIS** [reply]
> 我来当一个终结者：出现这个问题一般是两种情况，第一个没有配置路径映射。即没有按照作者说明，把/var/run/docker.sock映射到/var/run/docker.sock。 第二个是配置了映射，但是没权限。
> 
> 1）路径映射有3种办法。 一个是通过群晖套件的社区版资源下载，这里是共享好了。 第二个是通过群晖的docker安装，但是ssh到后台，安装的时候自定义好映射路径。 第三个就是在群晖的docker已经配置好了，不想删除重建，然而可视化更改路径映射不能选择/var/run/docker.sock。
> 
> 我是第三种。这里就要有点骚操作了。 第一步 ssh后，用这个命令，给docker.sock创建一个符号连接，类似于快捷方式的玩意。 ln -s /var/run/docker.sock /volume2/docker/homepage/。 此时去docker/homepage的后台就能看到docker.sock文件了。 但是群晖的可视化还是无法查看此文件，更不能选中。
> 
> 骚操作来了，先在homepag随便新建一个文件叫docker.sock（先把刚刚的符号连接重命名一下，命名为1），然后停止原先的容器，可视化选择文件映射，文件就用刚刚新建的这个docker.sock。 然后保存，不要运行容器。 保存完毕后，删除这个docker.sock，再把原始的docker.sock（被重命名为1的文件）再名字改回来。 改完启动容器即可。
> 
> 2）配置完映射，还是报错，就是没权限。 那就sudo chmod 666 /var/run/docker.sock 给他个权限。 解决重启失效，就在群晖创建个触发任务，每次开机自动运行这个命令即可

我这里是没有权限，sudo chmod 666 /var/run/docker.sock是可以解决的，但是我把这个做成触发任务开机自动执行失败，提示chmod: cannot access '/var/run/docker.sock': No such file or directory，但我在计划任务里手动点一下运行又可以运行成功，大佬有什么头绪吗？
